Source/eventHorizon/EventHorizons.ts

```diff
@@ -57,32 +63,78 @@ export class EventHorizons extends IEventHorizons {
             const consumerTenant = tenantWithSubscriptions.tenant;
 
             for (const subscription of tenantWithSubscriptions.subscriptions) {
-                const executionContext = this._executionContext.forTenant(consumerTenant.value);
 
-                this._logger.debug(`Subscribing to events from ${subscription.partition} in ${subscription.stream} of ${subscription.tenant} in ${subscription.microservice} for ${consumerTenant} into ${subscription.scope}`);
+                const pbSubscription = this.createSubscriptionRequest(consumerTenant, subscription);
+                // no point in having error handler, retryWithPolicy gulps up the pushed errors
+                this.subscribeWithRetry(consumerTenant, subscription, pbSubscription, this.cancellation)
+                    .subscribe({
+                        complete: () => {
+                            this._logger.debug(`Event Horizon subscription complete.`);
+                        }
+                    });
+            }
+        }
+    }
 
-                const callContext = callContexts.toProtobuf(executionContext);
-                callContext.setHeadid(guids.toProtobuf(Guid.create()));
+    private subscribeWithRetry(tenant: TenantId, subscription: Subscription, pbSubscription: PbSubscription, cancellation: Cancellation): Observable<void> {
+        return retryWithPolicy(
+            this.subscribe(tenant, subscription, pbSubscription),
+            retryPipe(delay(1000)),
```

Comment:
Maybe we should have a delay that gets longer and longer (max ceiling 30 seconds or something?)

Reply:
Yeah we could, I didn't think much about it. Depending on the error coming from the Runtime, it might take up to 20s or so for the connection to actually fail and trigger this retry. Looking ath the C# SDK, we have a 5s timeout there so it could definitely be increased to 5s at least.